File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.py
def solution(g, p):
    lst = {}
    g_lst = {}
    answer = []
    for i in range(len(g)):
        if g[i] not in lst:
            lst[g[i]] = 0
            g_lst[g[i]] = []
        lst[g[i]] += p[i]
        g_lst[g[i]].append((p[i],i))
    lst = sorted(lst.items(), key=lambda x: x[1], reverse=True)

    for genre, _ in lst:
        print(genre)
        g_lst[genre].sort(key=lambda x: (-x[0], x[1]))

        for i in range(min(2,len(g_lst[genre]))):
            answer.append(g_lst[genre][i][1])
    return answer

File: 프로그래머스/3/42579. 베스트앨범/베스트앨범.js
function solution(gs, ps) {
    var answer = [];
    let gSort = new Map()
    const eg = {}
    for (let i = 0; i< gs.length; i++){
        gSort.set(gs[i], (gSort.get(gs[i])||0) + ps[i])
        if (!eg[gs[i]]) eg[gs[i]] = []
        eg[gs[i]].push({plays: ps[i], idx : i})
    }
    gSort = [...gSort].sort((a,b)=> b[1]-a[1])
    for (s of gSort){
        // console.log(s[0]) 'pop' , 'classic'
        for (let i=0; i< Math.min(2, eg[s[0]].length); i++){
            eg[s[0]].sort((a,b)=> b.plays - a.plays)
            answer.push(eg[s[0]][i].idx)
        }
    }
    return answer;
}